hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMOpenKeysDeleteRequest.java

```diff
@@ -188,4 +197,46 @@ private void updateOpenKeyTableCache(OzoneManager ozoneManager,
     }
   }
 
+  /**
+   * Validates open keys delete requests.
+   * We do not want to allow older clients to delete open keys in buckets which
+   * use non LEGACY layouts.
+   *
+   * @param req - the request to validate
+   * @param ctx - the validation context
+   * @return the validated request
+   * @throws OMException if the request is invalid
+   */
+  @RequestFeatureValidator(
+      conditions = ValidationCondition.OLDER_CLIENT_REQUESTS,
+      processingPhase = RequestProcessingPhase.PRE_PROCESS,
+      requestType = Type.DeleteOpenKeys
+  )
+  public static OMRequest blockDeleteOpenKeysWithBucketLayoutFromOldClient(
```

Comment:
This one is also not submitted by the client, so we probably don't need to block it.

Reply:
Yes, exactly, this is an internal call and not required to block.
cc: @JyotinderSingh